compiler/modules/front/expr.bal

```diff
@@ -1224,8 +1224,24 @@ function codeGenFunctionCallExpr(ExprContext cx, bir:BasicBlock bb, s:FunctionCa
     }
     bir:BasicBlock curBlock = bb;
     bir:Operand[] args = [];
+    s:Expr[] restArgs = [];
+    t:SemType? restParamType = func.signature.restParamType;
     foreach int i in 0 ..< expr.args.length() {
-        var { result: arg, block: nextBlock } = check codeGenArgument(cx, curBlock, expr, func, i);
+        if restParamType != () && i >= func.signature.paramTypes.length() - 1 {
```

Comment:
This is not as clean as it could be.

- Start off by checking whether you have too few/too many arguments; how many are rest of arguments, how many not.
- One loop over regular arguments (as before)
- Separate if else/loop to deal with rest arguments

Reply:
fixed